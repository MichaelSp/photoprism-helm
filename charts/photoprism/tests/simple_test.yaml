suite: Simple Test Suite
templates:
  - configmap.yaml
  - ingress.yaml
  - secret.yaml
  - service.yaml
  - statefulset.yaml
values:
  - ../values.yaml
tests:
  - it: configmap matches the snapshot
    template: configmap.yaml
    asserts:
      - matchSnapshot: {}
  - it: ingress matches the snapshot
    template: ingress.yaml
    set:
      ingress.enabled: true
    asserts:
      - matchSnapshot: {}
  - it: secret matches the snapshot
    template: secret.yaml
    asserts:
      - matchSnapshot: {}
  - it: service matches the snapshot
    template: service.yaml
    asserts:
      - matchSnapshot: {}
  - it: statefulset matches the snapshot
    template: statefulset.yaml
    asserts:
      - matchSnapshot: {}