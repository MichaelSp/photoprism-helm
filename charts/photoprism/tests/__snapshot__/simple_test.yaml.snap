configmap matches the snapshot:
  1: |
    apiVersion: v1
    data:
      PHOTOPRISM_DEBUG: "true"
      PHOTOPRISM_PUBLIC: "false"
      PHOTOPRISM_READONLY: "true"
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: photoprism
        app.kubernetes.io/version: 230625-ce
        helm.sh/chart: photoprism-0.2.1
      name: RELEASE-NAME-photoprism-env
  2: |
    apiVersion: v1
    data:
      PHOTOPRISM_DATABASE_DRIVER: sqlite
      PHOTOPRISM_ORIGINALS_PATH: /photoprism/originals
      PHOTOPRISM_STORAGE_PATH: /photoprism/storage
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: photoprism
        app.kubernetes.io/version: 230625-ce
        helm.sh/chart: photoprism-0.2.1
      name: storage
ingress matches the snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: photoprism
        app.kubernetes.io/version: 230625-ce
        helm.sh/chart: photoprism-0.2.1
      name: RELEASE-NAME-photoprism
    spec:
      rules: null
secret matches the snapshot:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: photoprism
        app.kubernetes.io/version: 230625-ce
        helm.sh/chart: photoprism-0.2.1
      name: RELEASE-NAME-photoprism-credentials
    stringData:
      PHOTOPRISM_ADMIN_PASSWORD: photoprism
      PHOTOPRISM_DATABASE_DSN: /photoprism/storage/photoprism.sqlite
service matches the snapshot:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: photoprism
        app.kubernetes.io/version: 230625-ce
        helm.sh/chart: photoprism-0.2.1
      name: RELEASE-NAME-photoprism
    spec:
      ports:
        - name: http
          port: 80
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: photoprism
      type: ClusterIP
statefulset matches the snapshot:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: photoprism
        app.kubernetes.io/version: 230625-ce
        helm.sh/chart: photoprism-0.2.1
      name: RELEASE-NAME-photoprism-set
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: photoprism
      serviceName: RELEASE-NAME-photoprism
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: photoprism
        spec:
          containers:
            - env:
                - name: PHOTOPRISM_HTTP_HOST
                  value: 0.0.0.0
                - name: PHOTOPRISM_HTTP_PORT
                  value: "2342"
              envFrom:
                - configMapRef:
                    name: RELEASE-NAME-photoprism-env
                - configMapRef:
                    name: storage
                - secretRef:
                    name: RELEASE-NAME-photoprism-credentials
              image: photoprism/photoprism:231011-ce
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: photoprism
              ports:
                - containerPort: 2342
                  name: http
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              volumeMounts:
                - mountPath: /photoprism/originals
                  name: originals
                  subPath: photos/family
                - mountPath: /photoprism/storage
                  name: storage
          securityContext: {}
          terminationGracePeriodSeconds: 5
